indico/modules/rb_new/controllers/user/landing.py

```diff
@@ -16,10 +16,52 @@
 
 from __future__ import unicode_literals
 
+from datetime import datetime, time
+
+import dateutil
+from flask import request, session
+
+from indico.modules.rb import rb_settings
 from indico.modules.rb.controllers import RHRoomBookingBase
+from indico.modules.rb.models.rooms import Room
 from indico.modules.rb_new.views.base import WPRoomBookingBase
+from indico.util.string import natural_sort_key
+from indico.web.util import jsonify
 
 
 class RHRoomBookingLanding(RHRoomBookingBase):
     def _process(self):
         return WPRoomBookingBase.display('landing.html')
+
+
+class RHNewRoomBookingNewBooking(RHRoomBookingBase):
+    def _process(self):
+        from indico.modules.rb.schemas import rb_type_schema
+
+        min_capacity = int(request.args['capacity']) if request.args.get('capacity') else None
+        room_name = request.args.get('title')
+        query = Room.query.filter(Room.is_active)
+        start_dt = dateutil.parser.parse(request.args['start_dt']) if request.args.get('start_dt') else None
+        end_dt = dateutil.parser.parse(request.args['end_dt']) if request.args.get('end_dt') else None
+
+        if start_dt and end_dt:
+            selected_period_days = (end_dt - start_dt).days
+        else:
+            selected_period_days = 1
+
+        if min_capacity is not None:
+            query = query.filter(Room.capacity >= min_capacity)
+        if room_name:
+            query = query.filter(Room.name.ilike('%{}%'.format(room_name)))
+
+        rooms = []
+        for room in query:
+            booking_limit_days = room.booking_limit_days or rb_settings.get('booking_limit')
+            if booking_limit_days is not None and selected_period_days > booking_limit_days:
+                continue
+            rooms.append(room)
+
+        rooms = sorted(rooms, key=lambda r: natural_sort_key(r.full_name))
+        my_rooms = [room for room in Room.find_all() if room.is_owned_by(session.user)]
+        return jsonify(success=True, rooms=rb_type_schema.dumps(rooms).data,
+                       my_rooms=rb_type_schema.dumps(my_rooms).data)
```

Comment:
You're not yet filtering by availability of the room ("time slot" in the issue). You're just returning rooms that fit the capacity, name and max. booking length criteria.

Reply:
I would just return `jsonify(rb_type_schema.dumps(rooms).data)` (i.e. a list of rooms). No need to return 'my_rooms'.